docs/api/config.rst

```diff
@@ -12,7 +12,11 @@ To get your own config file, use the provided utility function
     config.copy_file()
 
 This will put a ``config.yaml`` in your current directory for you to edit. For more information about the YAML
-standard, visit https://yaml.org/spec/1.2/spec.html
+standard, visit https://yaml.org/spec/1.2/spec.html.
+
+.. warning:: PyYAML does not support conversion of scientific notation formatted floats into floats, so if you specify
```

Comment:
Ah, this is a really great warning to have, since it results in a pretty hard to catch error. I always wonder if there's something better than YAML here, perhaps just a user *.py file?

Reply:
that's how Django handles its configuration, and it's really nice because the user has a lot more control over what the values are, but the downside is that once you import the file, nothing can be changed. So there wouldn't be true lazy-loading and you'd have to restart the python instance if you change anything in the config after importing it.

Reply:
Although Django has a special lazy loader apparently ([see here](https://stackoverflow.com/a/14227305)). Overall, though, there isn't a great consensus on what configuration files to use- some people say that using python files for settings is very vulnerable but others say its preferable since you're writing pure python out of the get-go instead of converting and risking errors there. Plus users probably already know python but may not know the YAML standard (or similar) as well.

Reply:
Another aspect to consider is the actual use of our config files. Ideally all of the API methods should work without any configuraiton from the disk. In other words, every method should have some form of arguments that allow full control. The use of the config file, as far as I understand, is to simplify the process of doing a one-shot full pipeline. I can appreciate this point too.

We could probably end up with quite a few use-cases where there's no need for configuration on a settings file- but rather using arguments for the specification.